src/backend/base/langflow/components/git/gitfile.py

```diff
@@ -0,0 +1,186 @@
+            build_config["selected_files"]["options"] = ["Select branch first"]
+            build_config["selected_files"]["value"] = []
+            return build_config
+
+        if field_name in {"repository_url", "branch"}:
+            branches = await self.get_branches(self.repository_url)
+            build_config["branch"]["options"] = branches
+            if field_name == "repository_url":
+                build_config["branch"]["value"] = branches[0] if branches else "main"
+
+        if (
+            self.branch
+            and self.branch not in ["Enter repository URL first"]
+            and field_name in {"repository_url", "branch", "selected_files"}
+        ):
+            try:
+                async with self.temp_git_repo() as temp_dir:
+                    await asyncio.get_event_loop().run_in_executor(
+                        None,
+                        lambda: git.Repo.clone_from(
+                            self.repository_url, temp_dir, branch=self.branch, depth=1, single_branch=True
+                        ),
+                    )
+
+                    file_list = await self.get_repository_files(temp_dir)
+                    if not file_list:
+                        self.log("No files found in repository")
+                        build_config["selected_files"]["options"] = ["No files found in branch"]
+                        build_config["selected_files"]["value"] = []
+                    else:
+                        self.log(f"Found {len(file_list)} files")
+                        build_config["selected_files"]["options"] = file_list
+
+            except git.exc.GitCommandError as e:
+                error_msg = f"Git error: {e!s}"
+                self.log(error_msg)
+                build_config["selected_files"]["options"] = ["Error accessing branch"]
+                build_config["selected_files"]["value"] = []
+            except OSError as e:
+                error_msg = f"Error listing files: {e!s}"
+                self.log(error_msg)
+                build_config["selected_files"]["options"] = ["Error listing files"]
+                build_config["selected_files"]["value"] = []
+
+        return build_config
+
+    async def get_files_content(self) -> list[Data]:
+        if not self.repository_url:
+            return [Data(data={"error": "Please enter a repository URL"})]
+        if not self.branch or self.branch == "Enter repository URL first":
+            return [Data(data={"error": "Please select a branch"})]
+        if not self.selected_files:
+            return [Data(data={"error": "Please select at least one file"})]
+
+        try:
+            async with self.temp_git_repo() as temp_dir:
+                await asyncio.get_event_loop().run_in_executor(
+                    None,
+                    lambda: git.Repo.clone_from(
+                        self.repository_url, temp_dir, branch=self.branch, depth=1, single_branch=True
+                    ),
+                )
```

Comment:
Never use asyncio.get_event_loop() because there may not be a running event loop which will raise an error. Always use `asyncio.to_thread`.